extensions/arc/deployment/src/test/java/io/quarkus/arc/test/interceptor/staticmethods/InterceptedStaticMethodTest.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.arc.test.interceptor.staticmethods;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.reflect.Modifier;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InterceptorBinding;
+import javax.interceptor.InvocationContext;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.opentest4j.AssertionFailedError;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class InterceptedStaticMethodTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(InterceptMe.class, Simple.class, SimpleInterceptor.class));
+
+    @Test
+    public void testInterceptor() {
+        assertEquals("OK:PONG", Simple.ping("pong"));
+        Simple.pong();
+        assertEquals(42.0, Simple.testDouble(2.0));
+        assertEquals(1, SimpleInterceptor.VOID_INTERCEPTIONS.get());
+    }
+
+    public static class Simple {
+
+        @InterceptMe
+        public static String ping(String val) {
+            return val.toUpperCase();
+        }
+
+        @InterceptMe
+        static void pong() {
+        }
+
+        @InterceptMe
+        protected static Double testDouble(Double val) {
```

Comment:
When I said to try `double` and `long` primitives that's not what I meant ;) I mean, your ASM code looks right, but this is not testing it ;)

Reply:
Ah, I'm boxing the params automatically in my head ;-)